build_defs/internal/kotlin.bzl

```diff
@@ -36,20 +36,49 @@ def kt_jvm_and_js_library(
     kt_jvm_library(
         name = name,
         srcs = srcs,
-        deps = deps,
+        deps = [_to_jvm_dep(dep) for dep in deps],
     )
 
-    kt_js_library(
-        name = "%s-js" % name,
-        srcs = srcs,
-        deps = [_to_js_dep(dep) for dep in deps],
-    )
+    # Disabled while https://github.com/bazelbuild/rules_kotlin/issues/219 is unfixed.
+    # kt_js_library(
+    #    name = "%s-js" % name,
+    #    srcs = srcs,
+    #    deps = [_to_js_dep(dep) for dep in deps],
+    #)
+
+def _to_jvm_dep(dep):
+    return dep
 
 def _to_js_dep(dep):
     last_part = dep.split("/")[-1]
 
+    if (dep.find("@maven") == 0):
+        return _kt_js_import_for_thirdparty(dep)
+
     index_of_colon = dep.find(":")
     if (index_of_colon == -1):
         return dep + (":%s-js" % last_part)
     else:
         return dep + "-js"
+
+def _kt_js_import_for_thirdparty(thirdparty_dep):
+    name = "unknown"
+    version = ""
+    maven_name = "unknown"
+    if (thirdparty_dep == "@maven//:org_jetbrains_kotlinx_kotlinx_coroutines_core"):
```

Comment:
We're going to keep running into problems with maintaining copybara as well as two separate macro sets I can see. Here's a proposal:

Why not create a directory at the WORKSPACE root that mirrors g3, e.g.

third_party/kotlin/kotlinx/kotlinx_coroutines

Drop a BUILD file in there with an alias to @maven//:org_jetbrains_kotlinx_kotlinx_coroutines_core

Then we could use //third_party/kotlin/kotlinx/kotlinx_coroutines style deps in GitHub, import it without any rewrites by copybara needed, and just exclude the third_party/ directory from being copied?

Reply:
I like it! Let's do it in another PR though?

Reply:
Also, we could just bundle the `.jar` files themselves into our repo, and be able to skip all the nonsense with maven?  This would be even closer to how it works internally.

Reply:
I like both proposals :)